parserator_web/static/js/index.js

```diff
@@ -1,2 +1,47 @@
 /* TODO: Flesh this out to connect the form to the API and render results
    in the #address-results div. */
+$(document).ready(function() {
+   const $form = $('#js-parse-form')
+   const $addressResults = $('#js-address-results')
+   const $errorsEl = $('#js-errors')
+
+   $form.on('submit', (e) => {
+      e.preventDefault()
+      $errorsEl.html('').hide()
+
+      const address = $('#js-address').val()
+
+      if (!address) {
+         $addressResults.hide()
+         $errorsEl.text('Please enter a valid address')
+         $errorsEl.show()
+         return
+      }
```

Comment:
There is broad browser support for built-in validation these days. How about a small refactor to prevent empty submissions using the `input` element, instead of JavaScript? https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#using_built-in_form_validation

Reply:
Ah yes, a much simpler solution. Thanks for this as I do like to lean on built-in HTML features as much as possible. I've made this change.